### ISSUE TITLE
TD ignores the fact that I have already downloaded the video, and ignorantly tries to get the video again.

### ISSUE THREAD

**Issue by victorxstc** [telegramdesktop/tdesktop#28679]
### Steps to reproduce

1. On a popular Telegram channel, find a video.
2. Do **_NOT play_** the video. Do not click on the center of the video. 
3. Just **_download_** it by clicking the vertical arrow in the top left corner of the movie. 
4. Once the download is finished, now click on the center of the video to play the video on Telegram.

### Expected behaviour

Telegram Desktop should play the video immediately from the hard disk. The video is ALREADY downloaded and available locally to play instantly. This correct behavior used to happen until last month (there was no such a bug before this server-side resolution adjustment).

### Actual behaviour

Instead, Telegram Desktop doesn't understand that it has already downloaded the local version of the movie, connects again to the server (a lot of waiting), trying to stream the video from the server from scratch (a video that is already available locally).

This behavior started after Telegram Android started to adjust the video resolution server-side based on the user's connection speed. Now, Telegram Android doesn't even allow the user to "download" movies (as files, without opening them). It only allows them to "play" movies by opening the movies.

However, T Desktop has not caught on yet, and has both features of "download" and "play" with the latter being incomplete and buggy.

**Solution:**

Please either delete the "download" feature, or fix the bug of the "play" feature so that it first checks whether or not the video is downloaded and available locally, before even wanting to connect to the server and request streaming.

### Operating system

Windows 10

### Version of Telegram Desktop

5.8.2 x64 (the problem existed in ALL previous versions since a couple of months ago)

### Installation source

Static binary from official website

**Comment by FlossyMaze** [comment]
I was unable to reproduce this, does the downloaded file show up in Telegram Downloads? (Settings > Advanced > Downloads)

Tested on: Telegram Desktop 5.8.2 x64
OS Name: Microsoft Windows 10 Home Single Language
OS Version: 10.0.19045 N/A Build 19045

**Comment by victorxstc** [comment]
Yes, the video is in the Downloads list (Settings > Advanced > Downloads).
I should add that I have changed the default directory of downloads, and also the default directory of cache. Perhaps this "play" feature checked the original download directory, while it should check the download and cache directories that I have set.
Maybe this is why you couldn't reproduce it.

Besides the one I reported yesterday, now I found another funny bug that shows that the "download" and "play" features are independent of each other and don't recognize each other. I think I should create a new bug report for it, but its gist is this:

On a slow internet (where the video lasts a few seconds to download), I clicked the download arrow, and while it was being downloaded, I clicked the "play" triangle on the same video. I watched the whole video to the end using the "play" feature (the whole video is streamed). Then I closed the "player" and surprised to see that the "download" of the very same video is just midway (not downloaded completely).

**Comment by victorxstc** [comment]
The issue is there, and I didn't solve it. Actually, it is extremely annoying. An easy way to reproduce it is this:

1. Open a Telegram channel with video clips.
2. Download a number of clips only by clicking the downward arrow ("Download") that exists in the upper left corner of each video's thumbnail. Do **_NOT_** use the "Play" triangle in the middle of the thumbnail.
3. By clicking the downward arrow, Telegram downloads the video to its local folder (not its cache, but its storage).
4. Wait a little bit till the download is finished. When it is finished, the downward arrow will disappear, showing that the video is locally stored.

NOW it gets tricky:

5. Once Telegram downloaded all videos to local storage, **disconnect the computer from the internet**.

7. Now try playing the videos in Telegram: click the "Play" icon in the middle of the thumbnail in your Telegram Desktop. 
The BUG happens:

7. Instead of playing the already available video, Telegram Desktop will try to connect to the server and STREAM the video, i.e., the very same video that it has ALREADY DOWNLOADED. It can't understand that it does already have the whole video file in its local storage! 

8. Keep disconnected from the internet, Telegram will never ever be able to play the very file it has downloaded locally!! It will keep turning, waiting for streaming.

---------------------------------------------------------------------

What I expected to happen? What is intuitive and user-friendly? 

This: In older versions of Telegram Desktop, clicking the "Play" icon would simply check the local storage, and upon finding the video, simply open and play the locally stored video. Nevertheless, after this "streaming update" of Telegram (with its low and high video quality etc.), new versions ONLY work when connecting to server and stream the video, even when Telegram already does have downloaded and stored the very same video on its local storage address. They will play only and ONLY from Telegram Desktop's cache, not from its local storage.


I don't understand this very counter-intuitive behavior. If the download option is obsolete for videos, the downward "Download" arrow should not be there for videos, and the download functionality should be disabled for videos. If the download option is available and working for videos, then if the user has already downloaded the whole video, why should the user STREAM it again?

**Comment by liuhqsir** [comment]
It is indeed a bug, and it still exists in the latest 6.2.4 version.
This is a screen recording of me reproducing the issue.  
https://youtu.be/ur-pvv86c9c

**Comment by victorxstc** [comment]
> It is indeed a bug, and it still exists in the latest 6.2.4 version. This is a screen recording of me reproducing the issue. https://youtu.be/ur-pvv86c9c

Glad to see it finally reproduced. Thanks.

**Comment by victorxstc** [comment]
I wonder if Telegram wants to fix this bug or if it is intentional behavior. If it is intentional, it is definitely very poor design. If it is a bug (which @liuhqsir agrees), then I would much appreciate it if you could kindly fix it. I have reported it 2 years ago, and it still remains in the past 1.5 years.
